pennylane/ops/functions/equal.py

```diff
@@ -33,26 +34,56 @@ def equal(
 ):
     r"""Function for determining operator or measurement equality.
 
+    .. Warning::
+
+        The compare method does **not** check if the matrix representation
+        of a :class:`~.Hermitian` observable is equal to an equivalent
+        observable expressed in terms of Pauli matrices, or as a
+        linear combination of Hermitians.
+        To do so would require the matrix form of Hamiltonians and Tensors
+        be calculated, which would drastically increase runtime.
+
+        The kwargs ``check_interface`` and ``check_trainability`` can only be set when
+        comparing ``Operation`` objects. Comparisons of ``MeasurementProcess`` or ``Observable``
+        objects will use the defualt value of ``True`` for both, regardless of what the user
+        specifies when calling the function.
+
     Args:
         op1 (.Operator or .MeasurementProcess): First object to compare
         op2 (.Operator or .MeasurementProcess): Second object to compare
-        check_interface (bool, optional): Whether to compare interfaces. Default: ``True``
-        check_trainability (bool, optional): Whether to compare trainability status. Default: ``True``
+        check_interface (bool, optional): Whether to compare interfaces. Default: ``True``. Can only be set by user when comparing ``Operations``.
+        check_trainability (bool, optional): Whether to compare trainability status. Default: ``True``. Can only be set by user when comparing ``Operations``.
         rtol (float, optional): Relative tolerance for parameters
         atol (float, optional): Absolute tolerance for parameters
 
     Returns:
         bool: ``True`` if the operators or measurement processes are equal, else ``False``
 
-    **Example**
+    **Examples**
```

Comment:
Minor: we conventionally have `Example` for this section across the code base.
```suggestion
    **Example**
```